core/hmap.go

```diff
@@ -0,0 +1,82 @@
+package core
+
+import (
+	"errors"
+)
+
+type HashMap map[string]string
+
+func (h HashMap) Get(k string) (*string, bool) {
+	value, ok := h[k]
+	if !ok {
+		return nil, false
+	}
+	return &value, true
+}
+
+func (h HashMap) Set(k, v string) (*string, bool) {
+	value, ok := h[k]
+	if ok {
+		oldValue := value
+		h[k] = v
+		return &oldValue, true
+	}
+
+	h[k] = v
+	return nil, false
+}
+
+func hashMapBuilder(keyValuePairs []string, currentHashMap HashMap) (HashMap, int64, error) {
+	var hmap HashMap
+	var numKeysNewlySet int64
+
+	if currentHashMap == nil {
+		hmap = make(HashMap)
+	} else {
+		hmap = currentHashMap
+	}
+
+	iter := 0
+	argLength := len(keyValuePairs)
+
+	for iter <= argLength-1 {
+		if iter >= argLength-1 || iter+1 > argLength-1 {
+			return hmap, -1, errors.New("mismatch in number of fields and values provided to HSET")
```

Comment:
Error message shouldn't be associated with 1 command type ideally. If we're planning to allow this class to used by other command types like `HMSET, HPERSIST` etc.